### DISCUSSION TITLE
Profiling offheap memory leaks - can't make it work,

### DISCUSSION THREAD

**Discussion by cornelcreanga** [async-profiler/async-profiler#858] [General]
My Spark executors are crashing under heavy load due to too much native memory allocation (they are killed by k8 for exceeding quota requests). I suspect a memory leak and I wanted to try the approach described here:
https://github.com/async-profiler/async-profiler/blob/malloc/README.md#finding-native-memory-leaks

I've downloaded https://profiler.tools/builds/async-profiler-2.1-malloc-linux-x64.tar.gz and decompress it to /home/cornel/async-profiler-malloc but I can't make it work

The command

```LD_PRELOAD=/home/cornel/async-profiler-malloc/build/libasyncProfiler.so java -agentlib:/home/cornel/async-profiler-malloc/build/libasyncProfiler.so=start,event=nativemem,file=malloc.jfr -cp . com/creanga/playground/spark/TestProfiler```

breaks with

```Could not find agent library /home/cornel/async-profiler-malloc/build/libasyncProfiler.so on the library path, with error: lib/home/cornel/async-profiler-malloc/build/libasyncProfiler.so.so: cannot open shared object file: No such file or directory```

I tried with -agentlib:asyncProfiler=start, it also does not work.

Environment: java version "17.0.6" 2023-01-17 LTS, Ubuntu 22.04.3 LTS

Thanks a lot.
Cornel.

**Comment by apangin** [comment]
Try replacing  `-agentlib:` with `-agentpath:`

`-agentlib` expects a short library name (`asyncProfiler`), while `-agentpath` works with a full path name to `.so` library.

**Comment by cornelcreanga** [comment]
Thanks, it works.
